references/sdk.md

```diff
@@ -175,6 +175,9 @@ add_dataset(self, data: Union[dict, substra.sdk.schemas.DatasetSpec])
 ```
 
 Create new dataset asset and return its key.
+In debug mode, add the following key:`substra.sdk.DEBUG_OWNER` to the metadata,
+the value becomes the node owner of the data, and all tuples using this data
+have their worker set to this node.
```

Comment:
Is this only for datasets or shouldn't this mechanism work for all assets?

Reply:
Which assets are you thinking about ?

It is set on the dataset, and then if:
- an objective
- a traintuple
- a composite traintuple
- a testtuple
uses this dataset, then their worker is set to that value, like in the backend.  

The alternative would be to add this to the data samples, notably because if the objective is not linked to the dataset then we cannot get the right worker.  

The problem is that linking it to the datasamples means that whenever we link a dataset to a data sample, we have to check that the worker of the dataset is the same for all its data samples, and a dataset with no data samples (at creation) has no worker. It may be the right way to do it, did not find the equivalent in the backend (it is in the go code, I had trouble understanding how it worked.)

Reply:
I'm sorry I wasn't clear. If I'm not mistaken, the "real" substra put an owner field on all objects. My thinking was that we should also allow to setup this owner field for all objects purely for consistency even though we'll only use the value from the dataset object.

Reply:
Done, you can change the 'creator' for all assets (called 'owner' for the dataset) except for the data samples since they have no metadata.